### DISCUSSION TITLE
Strange boundary out error

### DISCUSSION THREAD

**Discussion by konmenel** [DualSPHysics/DualSPHysics#215] [DUALSPHYSICS_V5.2]
Hello,

I has simulating a case of an impact of a wave on a rectangular floating using Chrono a few days ago using a small kernel radius (h/dp=1.04), and I got an unexpected `JSphGpuSingle::AbortBoundOut` exception. Here is the message that I got:

```text
*** ERROR: Some boundary particle was excluded. ***
TimeStep: 10.913914  (Nstep: 4693196)
Total boundary: 4716  (fixed=0  moving=0  floating=4716)
Excluded for: position=4716  rhop=0  velocity=0
 

*** Exception (JSphGpuSingle::AbortBoundOut) at DualSPHysics/source/JSph.cpp:2601
Text: Fixed, moving or floating particles were excluded. Check VTK file Error_BoundaryOut.vtk with excluded particles.


Finished execution (code=1).
```

The strange thing is that the error is about the position of the floating however the line indicating which limit was exceeded is missing from the error, i.e.
```text
Some boundary particle exceeded the +Z limit (top limit) of the simulation domain.
```
Also, the floating was nowhere near the edges of the domain in the last Part file and I cannot see anything wrong with its movement.

Moreover, when I opened the `Error_BoundaryOut.vtk` file in Paraview nothing was rendering even though the file size was 254.9 KB. So I did some extra digging and I realized that the reason why Paraview was not rendering the file was because the position of all particles in the file were not a number (NaN).

The error happens a bit after the wave impacted the floating. I suspect there is something to do with Chrono, maybe during the update there is a division by 0 or something like that, leading to a NaN in the particle position, however I can't be certain.

I should note that I simulated this case before with different kernel sizes and everything was fine, only this case breaks.

I am still trying to debug this, but do you have any thoughts on what might be happening or point me to some direction on figuring this out?

Thank you

**Comment by EnglishA1** [comment]
Hi,

Have you tried manually setting your z direction domain limits at the end of the xml file? if they are set to the default option then gencase will shrink the max and min values from the initial definition in order to save memory and you can then have boundary particles trying to leave the domain. It might be that case that the amount of shrinkage is dependent on the kernel size and your smaller choice of h/dp value results in a smaller domain. You can increase the domain size by adding default +100% or an actual number to the posmax command at the bottom of the xml.

I hope this helps to find your problem

**Comment by konmenel** [comment]
Hi,

Thanks for the reply.

I manually set the z limits because I had an issue before with a flap wavemaker going out of bounds. However, the floating sits on top of a wall in the middle of the domain in the z direction. The expected movement of the floating (from past simulations and experiments) is tiny compared to the domain of the simulations so I don't expect it to reach any of the limits. Moreover, I see no problem with the motion in the last output file (basically 0 motion) and suddenly it breaks.

The issue with the error is that there is no context on what limit was exceeded (x,y or z) which should always be there according to the source code. One of the if statements, [lines 2528-2533 ](https://github.com/DualSPHysics/DualSPHysics/blob/ecf3e9a98fcc3957840dd11b9973c5c60e134065/src/source/JSph.cpp#L2528)in function `JSph::AbortBoundOut`, must be true since I am not using periodic boundary conditions. However, in my case all of those are false.
